### ISSUE TITLE
PicklingError when parallel sampling

### ISSUE THREAD

**Issue by fonnesbeck** [pymc-devs/pymc#660]
While running the `logistic` example, I get a PicklingError when trying to use the `njobs=2` argument. Running from current master.

```
In [8]: with model:
   ...:     step = pm.NUTS()
   ...:     trace = pm.sample(1000, step, njobs=2)
   ...:     
/usr/local/Cellar/python3/3.4.2_1/Frameworks/Python.framework/Versions/3.4/lib/python3.4/importlib/_bootstrap.py:321: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility
  return f(*args, **kwds)
---------------------------------------------------------------------------
PicklingError                             Traceback (most recent call last)
<ipython-input-8-d0bbceeef8a7> in <module>()
      1 with model:
      2     step = pm.NUTS()
----> 3     trace = pm.sample(1000, step, njobs=2)
      4 

/Users/fonnescj/GitHub/pymc/pymc/sampling.py in sample(draws, step, start, trace, chain, njobs, tune, progressbar, model, random_seed)
     86         sample_args = [draws, step, start, trace, chain,
     87                        tune, progressbar, model, random_seed]
---> 88     return sample_func(*sample_args)
     89 
     90 

/Users/fonnescj/GitHub/pymc/pymc/sampling.py in _mp_sample(njobs, args)
    207 def _mp_sample(njobs, args):
    208     p = mp.Pool(njobs)
--> 209     traces = p.map(argsample, args)
    210     p.close()
    211     return merge_traces(traces)

/usr/local/Cellar/python3/3.4.2_1/Frameworks/Python.framework/Versions/3.4/lib/python3.4/multiprocessing/pool.py in map(self, func, iterable, chunksize)
    258         in a list that is returned.
    259         '''
--> 260         return self._map_async(func, iterable, mapstar, chunksize).get()
    261 
    262     def starmap(self, func, iterable, chunksize=None):

/usr/local/Cellar/python3/3.4.2_1/Frameworks/Python.framework/Versions/3.4/lib/python3.4/multiprocessing/pool.py in get(self, timeout)
    597             return self._value
    598         else:
--> 599             raise self._value
    600 
    601     def _set(self, i, obj):

/usr/local/Cellar/python3/3.4.2_1/Frameworks/Python.framework/Versions/3.4/lib/python3.4/multiprocessing/pool.py in _handle_tasks(taskqueue, put, outqueue, pool, cache)
    381                     break
    382                 try:
--> 383                     put(task)
    384                 except Exception as e:
    385                     job, ind = task[:2]

/usr/local/Cellar/python3/3.4.2_1/Frameworks/Python.framework/Versions/3.4/lib/python3.4/multiprocessing/connection.py in send(self, obj)
    204         self._check_closed()
    205         self._check_writable()
--> 206         self._send_bytes(ForkingPickler.dumps(obj))
    207 
    208     def recv_bytes(self, maxlength=None):

/usr/local/Cellar/python3/3.4.2_1/Frameworks/Python.framework/Versions/3.4/lib/python3.4/multiprocessing/reduction.py in dumps(cls, obj, protocol)
     48     def dumps(cls, obj, protocol=None):
     49         buf = io.BytesIO()
---> 50         cls(buf, protocol).dump(obj)
     51         return buf.getbuffer()
     52 

PicklingError: Can't pickle <function TensorType.values_eq_approx_remove_nan at 0x10d587598>: attribute lookup values_eq_approx_remove_nan on theano.tensor.type failed
```

**Comment by kyleam** [comment]
I can reproduce this.  This error also occurs with `psample` from before the backend merge (tested with commit aa16ce5b0691ab92edfd3b5a6746b7eb7cadfa58).

**Comment by kyleam** [comment]
Above I was referring to the logistic example. I get the same error when running hierarchical.py with `njobs=2`, but running normal.py does not produce the error.

**Comment by kyleam** [comment]
[These changes](https://github.com/kyleam/Theano/commit/b447ed07abf6abc4892119f15f5ce89f9e98ed55) to Theano fix the issue for me.  I'll test it out a bit more and then open a PR over there.

@fonnesbeck Can you confirm that those changes work for on your end?

**Comment by davidia** [comment]
Thanks, fixed it for me too, did you open a PR?

**Comment by kyleam** [comment]
No, I did not.  Thanks for the reminder.

**Comment by kyleam** [comment]
This should now be fixed in the master branch of theano (de93a12fd3cf).
